node/pkg/cni/token_watch.go

```diff
@@ -101,3 +82,47 @@ current-context: calico-context`
 	}
 	logrus.WithField("path", kubeconfigPath).Info("Wrote updated CNI kubeconfig file.")
 }
+
+func createCNIToken(kubecfg *rest.Config) string {
+	clientset, err := kubernetes.NewForConfig(kubecfg)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to create clientset for CNI kubeconfig")
+		return "invalid-token"
+	}
+
+	tokenRequestSupported := func() bool {
+		tokenOnce.Do(func() {
+			resources, err := clientset.Discovery().ServerResourcesForGroupVersion("v1")
+			if err != nil {
+				return
+			}
+			for _, resource := range resources.APIResources {
+				if resource.Name == "serviceaccounts/token" {
+					tokenSupported = true
+					return
+				}
+			}
+		})
+		return tokenSupported
+	}
+
+	validity := int64(defaultCNITokenValiditySeconds)
+	tr := &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			Audiences:         []string{"kubernetes"},
+			ExpirationSeconds: &validity,
+		},
+	}
+
+	tokenRequest, err := clientset.CoreV1().ServiceAccounts(metav1.NamespaceSystem).CreateToken(context.TODO(), "calico-node", tr, metav1.CreateOptions{})
```

Comment:
A note to myself - this gives us a great opportunity to finally split out the calico-node serviceaccount into "calico-node" and "calico-cni-plugin", giving more precise permissions to each component. 

No need to do that in this PR, but a great bit of follow-on work. I'll raise an enhancement to track.

Reply:
https://github.com/projectcalico/calico/issues/5921

Reply:
I assume that this is now tracked in #5921 and hence can be resolved.